### ISSUE TITLE
EarlyJsError: Tried to access NativeModule "RNDeviceInfo" from the bridge. This isn't allowed in Bridgeless mode.

### ISSUE THREAD

**Issue by smfunder** [react-native-device-info/react-native-device-info#1681]
## Summary

| Version     |  14.0.1   |
| Affected OS | Android   |
| OS Version  | Android 13  |

## Current behavior

I have my setup of the app with newArch enabled, but I set it to bridgeless=false in load(bridgelessEnabled=false) in MainApplication.kt file.

I'm using Firebase to receive messages in headless mode by setting messaging().setBackgroundMessageHandler

When I force close the app (quite state) and I receive a high priority push notification making the app to wakeup with headless mode, I receive a crash that says the following:

terminating due to uncaught exception of type facebook::jni::JniException: com.facebook.react.common.JavascriptException: EarlyJsError: Tried to access NativeModule "RNDeviceInfo" from the bridge. This isn't allowed in Bridgeless mode., stack:
                                                                                                    anonymous@1:1597397
                                                                                                    loadModuleImplementation@1:140689
                                                                                                    guardedLoadModule@1:140212
                                                                                                    metroRequire@1:139842
                                                                                                    anonymous@1:1583073
                                                                                                    loadModuleImplementation@1:140689
                                                                                                    guardedLoadModule@1:140212
                                                                                                    metroRequire@1:139842
                                                                                                    anonymous@1:849034
                                                                                                    loadModuleImplementation@1:140689
                                                                                                    guardedLoadModule@1:140212
                                                                                                    metroRequire@1:139842
                                                                                                    anonymous@1:769309
                                                                                                    loadModuleImplementation@1:140689
                                                                                                    guardedLoadModule@1:140212
                                                                                                    metroRequire@1:139842
                                                                                                    anonymous@1:146921
                                                                                                    loadModuleImplementation@1:140689
                                                                                                    guardedLoadModule@1:140169
                                                                                                    metroRequire@1:139842
                                                                                                    global@1:139370

Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 19608 (mqt_v_js), pid 19509 

It is suppose that I'm running in bridgless mode OFF, but it is still saying it is bridgeless.

## Expected behavior

The app should not crash.

I even tried this PR/Fork: https://github.com/alanjhughes/react-native-device-info.git#@alanhughes/new-arch-support

But I'm receiving the same issue.

Any help?

**Comment by mikehardy** [comment]
> I have my setup of the app with newArch enabled, but I set it to bridgeless=false in load(bridgelessEnabled=false) in MainApplication.kt file.

> I'm using Firebase to receive messages in headless mode by setting messaging().setBackgroundMessageHandler

I happen to be the react-native-firebase maintainer as well. You don't want to set bridgeless=false if you are using setBackgroundMessageHandler because in react-native 0.76.x there is a bug in non-bridgeless mode that hasn't been fixed 

https://github.com/facebook/react-native/issues/47592#issuecomment-2482871724

So, what happens if you get rid of that `bridgeless=false` parameter? Does it work?

**Comment by smfunder** [comment]
@mikehardy thank you for your quick answer! I got it, so looks like when running setBackgroundMessageHandler it ends up running into bridgeless mode despite I put it disabled on the configuration? 

If I move to bridgeless, we have a very huge app that has like 120 libraries and some of the them like React Native Track Player may no support new arch + bridgeless, so the background task may work, but not the rest of the app.

Do you know if in RN 0.75 this issue would not exist? If so, I may be switching to 0.75 in order to continue, until I can run all the libraries in new arch+bridgeless mode.

**Comment by mikehardy** [comment]
I would stay on 75 if you need headlessJS to work flawlessly, otherwise I would consider that you are integration testing + keep it in bridgeless mode + and log issues you find (but don't log issues with bridgeless=false as I think that's just going to waste time when we need to get bridgeless working)

I think RNTP may work now with latest 0.76.3 - the author there was one of the primary drivers of fixes for headlessJS and at least with bridgeless I think it's working now.

In other words avoid the "may not support new arch + bridgeless" guess and run it that way to see, or stay on 0.75 in my opinion

**Comment by smfunder** [comment]
@mikehardy finally I was able to mostly run everything with bridgeless+new arch with RN 0.76.3. But I still facing a random crash in Android (https://github.com/software-mansion/react-native-reanimated/issues/6764)

**Comment by smfunder** [comment]
Closing this issue as I finally turn into RN 0.76.3 with New Arch + Bridgeless mode and I got a crash related (or not related to micro tasks) that was caused by the react-native-push-notification library in Android in release mode.
